src/image-alt-text/opportunityHandler.js

```diff
@@ -62,11 +64,58 @@ export async function syncAltTextSuggestions({ opportunity, newSuggestionDTOs, l
     }
   }
 }
-// TO-DO: Implement in https://jira.corp.adobe.com/browse/ASSETS-47371
-const getProjectedMetrics = () => ({
-  projectedTrafficLost: 3871,
-  projectedTrafficValue: 7355,
-});
+
+const getProjectedMetrics = async ({
+  images, auditUrl, context, log,
+}) => {
+  const CPC = 1; // $1
+  const PENALTY_PER_IMAGE = 0.01; // 1%
+
+  const finalUrl = await getRUMUrl(auditUrl);
+  const INTERVAL = 30; // days
+
+  const rumAPIClient = RUMAPIClient.createFrom(context);
+  const options = {
+    domain: finalUrl,
+    interval: INTERVAL,
+  };
+
+  const results = await rumAPIClient.query('traffic-acquisition', options);
+  log.info(`[${AUDIT_TYPE}]: RUM API results: ${JSON.stringify(results)}`);
+
+  const pageUrlToOrganicTrafficMap = results.reduce((acc, page) => {
+    acc[page.url] = {
+      organicTraffic: page.earned,
+      imagesWithoutAltText: 0,
+    };
+    return acc;
+  });
+  log.info(`[${AUDIT_TYPE}]: Page URL to organic traffic map: ${JSON.stringify(pageUrlToOrganicTrafficMap)}`);
+
+  images.forEach((image) => {
+    const fullPageUrl = new URL(image.pageUrl, auditUrl).toString();
+    if (pageUrlToOrganicTrafficMap[fullPageUrl]) {
+      pageUrlToOrganicTrafficMap[fullPageUrl].imagesWithoutAltText += 1;
+    } else {
+      log.error(`[${AUDIT_TYPE}]: Page URL ${fullPageUrl} not found in RUM API results`);
+    }
+  });
+  log.info(`[${AUDIT_TYPE}]: Page URL to organic traffic map after processing images: ${JSON.stringify(pageUrlToOrganicTrafficMap)}`);
+
+  const projectedTrafficLost = Object.values(pageUrlToOrganicTrafficMap)
+    .reduce(
+      (acc, page) => acc + (page.organicTraffic * PENALTY_PER_IMAGE * page.imagesWithoutAltText),
+      0,
+    );
+  log.info(`[${AUDIT_TYPE}]: Projected traffic lost: ${projectedTrafficLost}`);
```

Comment:
while handy for debugging during dev, please try to be minimalist in log statements. most of them are never read on production and we're trying to reduce our log volumes. please apply to all dev work / files.

Reply:
This is strictly for me for DEV now, I usually don't leave any log actually before I merge. But that might not be ideal either...

Do you have any guidelines on what would be important to have in the logs? Is there something needed for auditing purposes?

Reply:
i don't have a blanket rule, just that the mindset should be "i add the minimal log needed to help me identify a production issue or where necessary to extract metrics for log-based alerting"